path_roles_test.go

```diff
@@ -217,6 +223,8 @@ func TestRoleCreate(t *testing.T) {
 				}]`,
 			),
 			"application_object_id": "",
+			"sign_in_audience":      "",
+			"tags":                  []string{""},
 			"azure_groups":          "[]",
 			"persist_app":           false,
 		}
```

Comment:
Should we add coverage for some of the invalid cases to confirm the plugin behaves in a predictable way? For example, Azure tags are case-insensitive so creating a role with:
`vault write azure/roles/my-role-1 ... sign_in_audience=AzureADMyOrg tags="case","CASE"`

Results in this error when generating credentials:
```
Error reading local-secrets-azure/creds/my-role-1: Error making API request.

URL: GET http://127.0.0.1:8200/v1/local-secrets-azure/creds/my-role-1
Code: 500. Errors:

* 1 error occurred:
	* Request contains a property with duplicate values.
```

I know there are precedents in Vault that we sometimes prefer failing with a 500 instead of adding pre-conditions to return a client error. But it should be tested or documented so if we decide to let the Azure errors bubble up instead of validating the inputs it is deliberate.

Reply:
Good call. I added validation based on the docs here: https://learn.microsoft.com/en-us/entra/identity-platform/supported-accounts-validation

I also added some test cases for invalid tags and sign in audience to make sure it behaves correctly and we do get an error